### ISSUE TITLE
TestCaseBackgroundScreenBeatmap randomly dies

### ISSUE THREAD

**Issue by peppy** [ppy/osu#4499]
Click `[SetUp]` then `Start player loader` in that order until an exception occurs:

```
[runtime:error] 3/20/19 11:28:00 AM: Skin failed to load
[runtime:error] 3/20/19 11:28:00 AM: System.NullReferenceException: Object reference not set to an instance of an object.
[runtime:error] 3/20/19 11:28:00 AM: at osu.Game.Skinning.LegacySkin..ctor(SkinInfo skin, IResourceStore`1 storage, AudioManager audioManager, String filename) in /Users/dean/Projects/osu/osu.Game/Skinning/LegacySkin.cs:line 41
[runtime:error] 3/20/19 11:28:00 AM: at osu.Game.Skinning.LegacyBeatmapSkin..ctor(BeatmapInfo beatmap, IResourceStore`1 storage, AudioManager audioManager) in /Users/dean/Projects/osu/osu.Game/Skinning/LegacyBeatmapSkin.cs:line 13
[runtime:error] 3/20/19 11:28:00 AM: at osu.Game.Beatmaps.BeatmapManager.BeatmapManagerWorkingBeatmap.GetSkin() in /Users/dean/Projects/osu/osu.Game/Beatmaps/BeatmapManager_WorkingBeatmap.cs:line 136
```

```
[runtime:error] 3/20/19 11:28:06 AM: Step Start player loader triggered an error
[runtime:error] 3/20/19 11:28:06 AM: System.InvalidOperationException: Cannot Push to a non-loaded IScreen directly. Consider using Push instead.
[runtime:error] 3/20/19 11:28:06 AM: at osu.Framework.Screens.ScreenExtensions.<>c.<Push>b__0_1() in C:\projects\osu-framework-a4n7e\osu.Framework\Screens\IScreen.cs:line 55
[runtime:error] 3/20/19 11:28:06 AM: at osu.Game.Tests.Visual.TestCaseBackgroundScreenBeatmap.<PlayerLoaderSettingsHoverTest>b__11_0() in /Users/dean/Projects/osu/osu.Game.Tests/Visual/TestCaseBackgroundScreenBeatmap.cs:line 100
[runtime:error] 3/20/19 11:28:06 AM: at osu.Framework.Testing.Drawables.Steps.SingleStepButton.<.ctor>b__1_0() in C:\projects\osu-framework-a4n7e\osu.Framework\Testing\Drawables\Steps\SingleStepButton.cs:line 17
[runtime:error] 3/20/19 11:28:06 AM: at osu.Framework.Testing.Drawables.Steps.StepButton.PerformStep(Boolean userTriggered) in C:\projects\osu-framework-a4n7e\osu.Framework\Testing\Drawables\Steps\StepButton.cs:line 117
[runtime:error] 3/20/19 11:28:06 AM: at osu.Framework.Testing.Drawables.Steps.StepButton.OnClick(ClickEvent e) in C:\projects\osu-framework-a4n7e\osu.Framework\Testing\Drawables\Steps\StepButton.cs:line 93
[runtime:debug] 3/20/19 11:28:06 AM: ClickEvent(Left) handled by Start player loader.
```

**Comment by peppy** [comment]
Specific repro case has now been resolved; use `4e8945a` for future reproduction.

**Comment by peppy** [comment]
Closing this one. the `LegacySkin` part will likely be resolved by audiomanager consumption changes in #4873.
